core/Rename.cpp

```diff
@@ -132,18 +132,21 @@ namespace olympia
         }
 
         const auto & srcs = inst_ptr->getRenameData().getSourceList();
-        // decrement reference for store instruction to data register
-        if(inst_ptr->isStoreInst()) {
+        // decrement reference to data register
+        if(inst_ptr->isLoadStoreInst()){
```

Comment:
@danbone's fix is a good catch: noticing that the renaming and reclamation are not in sync

But there's something nagging me about all of this -- we're renaming `x0`.  Can you open a new issue that fixes that?

Specifically, in `renameInstructions_`, when iterating over the sources and destinations, you need to _skip_ renaming `x0`:  

```
if (num == 0 && rf == core_types::RF_INTEGER) { continue; }
```